### PR TITLE
Catch exceptions in reticulum events and add some more logging and stats

### PR DESCRIPTION
**Pull request by keianhzo**
Catch exceptions in reticulum events and add some more logging and stats

### CODE REVIEW

File: habitat/plan.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 pkg_name=hubs-discord-bot
 pkg_origin=mozillareality
 pkg_maintainer="Mozilla Mixed Reality <mixreality@mozilla.com>"
-pkg_version="0.0.2"
+pkg_version="0.0.5"
```
Comment (brianpeiris) [inline habitat/plan.sh:4]: You don't actually have to update this version with every push. The package is already versioned with a timestamp. I think we generally use this `pkg_version` when we need to signal some major backwards incompatibility issue, but even then it doesn't really apply to the discord bot.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hubs-discord-bot",
-  "version": "0.0.2",
+  "version": "0.0.5",
```
Comment (brianpeiris) [inline package.json:3]: Same here, I wouldn't bother changing these versions. It doesn't affect anything.

File: src/index.js [hunk 3]
```diff
@@ -267,120 +265,144 @@ async function establishBridging(hubState, bridges) {
...
+          lastPresenceMessages[discordCh.id] = lastPresenceMessages[discordCh.id].then(prev => prev.edit(msg));
+        } else if (kind === "depart") {
+          const msg = `${formatList(users)} left the Hubs room.`;
+          lastPresenceMessages[discordCh.id] = lastPresenceMessages[discordCh.id].then(prev => prev.edit(msg));
+        } else if (kind === "rename") {
+          lastPresenceMessages[discordCh.id] = lastPresenceMessages[discordCh.id].then(prev => prev.edit(formatRename(users[0])));
+        }
+      }
+    } catch (e) {
+      console.error(ts(`Error updrelaying presence update ${kind} in ${hubState.id}.`));
```
Comment (brianpeiris) [inline src/index.js:314]: typo?
```suggestion
      console.error(ts(`Error relaying presence update ${kind} in ${hubState.id}.`));
```

File: src/index.js [hunk 4]
```diff
@@ -267,120 +265,144 @@ async function establishBridging(hubState, bridges) {
...
+          lastPresenceMessages[discordCh.id] = discordCh.send(msg);
+        } else if (kind === "depart") {
+          const msg = `${formatList(users)} left the Hubs room.`;
+          lastPresenceMessages[discordCh.id] = discordCh.send(msg);
+        } else if (kind === "rename") {
+          lastPresenceMessages[discordCh.id] = discordCh.send(formatRename(users[0]));
+        }
+      }
+    } catch (e) {
+      console.error(ts(`Error relaying presence ${kind} in ${hubState.id}.`));
```
Comment (brianpeiris) [inline src/index.js:289]: Should all these `catch` logs also include the error object, or at least `e.name` or `e.message`?

**Review by brianpeiris — APPROVED** [review]
Minor comments. LGTM otherwise.

File: habitat/plan.sh (hunk 1 above)
Comment (keianhzo) [reply habitat/plan.sh:4]: Should I revert or it's already too late?

File: src/index.js (hunk 3 above)
Comment (keianhzo) [reply src/index.js:314]: Yep, thanks!

File: habitat/plan.sh (hunk 1 above)
Comment (brianpeiris) [reply habitat/plan.sh:4]: Ah, if you've already deployed this, it's too late. Not a big deal though, just a suggestion that you can avoid it next time.

**Pull request merged**